embeds/linux_desktop.go

```diff
@@ -0,0 +1,79 @@
+package embeds
+
+import (
+	"bytes"
+	_ "embed"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/anywherelan/ts-dns/util/lineread"
+
+	"github.com/anywherelan/awl/config"
+)
+
+//go:embed .desktop
+var desktopFile []byte
+
+const (
+	desktopFileMod  = 0764 // rwxrw-r--
+	desktopFileName = "awl.desktop"
+	desktopFileDir  = ".local/share/applications"
+)
+
+func EmbedDesktopFile(iconPath string) error {
+	homeDir := os.Getenv("HOME")
+	if len(homeDir) == 0 {
+		return errors.New("error: empty home dir")
+	}
+
+	ex, err := os.Executable()
+	if err != nil {
+		return fmt.Errorf("error: find executable path: %w", err)
+	}
+
+	replacements := map[string][]byte{
+		"{VERSION}":  []byte(config.Version),
+		"{ICONPATH}": []byte(iconPath),
+		"{EXECPATH}": []byte(ex),
+	}
+
+	filePath := filepath.Join(homeDir, desktopFileDir, desktopFileName)
+
+	file, err := os.Create(filePath)
+	if err != nil {
+		return fmt.Errorf("error: create file: %w", err)
+	}
+	defer func() {
+		if err := file.Close(); err != nil {
+			fmt.Printf("error: close file %v\n", err)
+		}
+	}()
+
+	err = lineread.Reader(bytes.NewReader(desktopFile), func(line []byte) error {
```

Comment:
I think it could be done easier with just for loop with replaces and then using `os.WriteFile`, without opening and closing file in a few steps

Something like this:

```go
	desktopFileCopy := bytes.Clone(desktopFile)
	for replName, replVal := range replacements {
		desktopFileCopy = bytes.ReplaceAll(desktopFileCopy, []byte(replName), replVal)
	}
	_ = os.WriteFile(filePath, desktopFileCopy, desktopFileMod)
```